### PR TITLE
add MUMPS to PETSc

### PR DESCRIPTION
**Pull request by migueldiascosta**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
sms - Linux centos linux 7.6.1810, AMD EPYC 7601 32-Core Processor, Python 2.7.5
See https://gist.github.com/80b483f9947599fb9b8b4d2638cf5c95 for a full test report.

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/af7c5f82e53076aed0b8d8213656c621 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 9 out of 9 (4 easyconfigs in this PR)
login2.cerberus.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/77a7647ae00e9dd578af8c5d4ad60682 for a full test report.

**Comment by lexming** [comment]
LGTM

File: easybuild/easyconfigs/p/PETSc/PETSc-3.12.4-foss-2019b.eb [hunk 1]
```diff
@@ -26,8 +26,7 @@ builddependencies = [('CMake', '3.15.3')]
 
 dependencies = [
     ('Boost', '1.71.0'),
-    ('METIS', '5.1.0'),
```
Comment (boegel) [inline easybuild/easyconfigs/p/PETSc/PETSc-3.12.4-foss-2019b.eb]: @migueldiascosta Removing this direct dependency on `METIS` is causing issues with `SLEPc` on top of this `PETSc`:

```
/user/gent/400/vsc40023/eb_phanpyscratch/CO7/skylake-ib/software/PETSc/3.12.4-foss-2019b/lib/libpetsc.so: error: undefined reference to 'METIS_ComputeVertexSeparator'
/user/gent/400/vsc40023/eb_phanpyscratch/CO7/skylake-ib/software/PETSc/3.12.4-foss-2019b/lib/libpetsc.so: error: undefined reference to 'METIS_NodeND'
collect2: error: ld returned 1 exit status
```

Comment (lexming) [reply easybuild/easyconfigs/p/PETSc/PETSc-3.12.4-foss-2019b.eb]: `METIS` should be loaded through the dependency on `MUMPS`, isn't it the case?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node3165.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/d0a6a6768c82f382a66426ed26543de7 for a full test report.

File: easybuild/easyconfigs/p/PETSc/PETSc-3.12.4-foss-2019b.eb (hunk 1 above)
Comment (lexming) [reply easybuild/easyconfigs/p/PETSc/PETSc-3.12.4-foss-2019b.eb]: I can see the errors of `ld` regarding the undefined references on `METIS` in my successful test, but EB ignores those errors. These seems more of a bug in `petsc` easyblock than in this easyconfig.

Comment (boegel) [reply easybuild/easyconfigs/p/PETSc/PETSc-3.12.4-foss-2019b.eb]: fixed in #9891

**Pull request merged**